### PR TITLE
Fix display behavior of metadata-only collections

### LINKED ISSUE #466: Fix display behavior of metadata-only collections
### Descriptive summary

- Center for Music Experiment
- Dept. of Music Concert Archive
- ECE Distinguished Lecturer Series
- Kelco Aerial Photograph Collection
- Scientific Voyages and Expedition Reports

Currently on staging, these collections are not behaving as expected. If I am on campus, I am seeing metadata-only behavior. 

### Expected behavior

- When on campus or using VPN (UCSD IP only), user should be able to view the metadata, see (images), or play files (audio/video), but should not be able to download (no download button either).

- When off campus, user will see the metadata only display with the banner on the object and generic thumbnails on grid view (no download button).

- When logged in as curators, user should be able to view the metadata, see (images), or play files (audio/video), and download the master files.

### Related work

#356

### PR DESCRIPTION
**Pull request by VivianChu**
Fixes #466 

#### Local Checklist
- [x] Tests written and passing locally?
- [x] Code style checked?
- [x] QA-ed locally?
- [x] Rebased with `master` branch?
- [ ] Configuration updated (if needed)?
- [ ] Documentation updated (if needed)?

#### What does this PR do?
Fix restricted thumbnail issue for search results page.
Fix pdf file download issue for Scientific Voyages and Expedition Reports collection.
Add hide access label function for local user in search/browse by collection page, single object and collection view for metadata-only collection.

References #466 

#### Deployment Instructions
It has been deployed to staging and tested by Gabriela, Longshou and me.

@ucsdlib/developers - please review

### CODE REVIEW

File: app/controllers/catalog_controller.rb [hunk 1]
```diff
@@ -321,8 +321,7 @@ def collection_search
 
   def metadata_only_collections(params)
     meta_colls = []
-    solr_params = {}
-    solr_params[:fq] = ["{!join from=collections_tesim to=id}#{metadata_only_fquery}"]
+    solr_params = { q: "{!join from=collections_tesim to=id}#{metadata_only_fquery}", fq: 'type_tesim:Collection', rows: 300 }
```
Comment (lsitu) [inline app/controllers/catalog_controller.rb]: Is there any side effects when merging the parameter :q with that :q param passing in hash params in the line#325? Maybe we can try to use something like 
`{ fq: ["{!join from=collections_tesim to=id}#{metadata_only_fquery}", 'type_tesim:Collection'], rows: 300 }`?

File: spec/features/catalog_spec.rb [hunk 2]
```diff
@@ -586,17 +613,60 @@
     expect(page).to have_content('Restricted View')
   end
 
-  scenario 'local user should see the grey generic thumbnail and restricted access info' do
+  scenario 'local user should not see the grey generic thumbnail and still see restricted access info' do
```
Comment (lsitu) [inline spec/features/catalog_spec.rb]: Could we change ` still see` to match it with the expected test result `expect(page).to_not have_content('Restricted View')`?

**Comment by lsitu** [comment]
@VivianChu It looks good to me. Just a couple of minor questions. 👍

File: spec/features/catalog_spec.rb (hunk 2 above)
Comment (VivianChu) [reply spec/features/catalog_spec.rb]: @lsitu - I changed the expected test result but forgot to change the scenario.  I'll update it.  Thanks for catching it.

File: app/controllers/catalog_controller.rb (hunk 1 above)
Comment (VivianChu) [reply app/controllers/catalog_controller.rb]: @lsitu - I don't know if there's any side effect.  I could try your suggestion and deploy to staging to test.  Last time, it has problem on staging with the old fq.

File: spec/features/catalog_spec.rb (hunk 2 above)
Comment (VivianChu) [reply spec/features/catalog_spec.rb]: @lsitu - I deployed the change with that suggested query to staging.  Other collections from #466 worked fine.  However, some mixed collection and some sample collections that Greg created didn't work.  For example - https://librarytest.ucsd.edu/dc/search?utf8=%E2%9C%93&q=visibility   As curator, I see access label = Restricted to UC San Diego use only, but if I click on that link I see access label = Some items restricted.  The old query also worked fast too.  I don't see any different in performance between the old and the new query.  Do you have any other suggestion that I can try?  Thanks

File: app/controllers/catalog_controller.rb (hunk 1 above)
Comment (lsitu) [reply app/controllers/catalog_controller.rb]: @mcritchlow I'll be off line tomorrow and I don't want to block the PR. Could you give it a pass if you don't see any issues with the query? I am not sure how merging of two :q parameter works in https://github.com/ucsdlib/damspas/pull/503/files#diff-956e46b33fb5307990c4e1a4b5fd86ccR325.

Comment (VivianChu) [reply app/controllers/catalog_controller.rb]: @lsitu - I reverted the change to my query and deployed to staging. The search results worked fine. Both collection view and search view access label matched. It’s hard to have a test for it since staging will return larger results then local test with only few records created. There are already some tests to catalog spec and dams-collection test for search result access label. Those tests passed for both of our queries. They just behaved differently on staging.

**Comment by mcritchlow** [comment]
This LGTM. The test coverage looks comprehensive and takes into account the various (there are so many..) edge cases I've seen discussed on the ticket. Nice work @VivianChu :+1:

**Comment by mcritchlow** [comment]
@hweng - Any concerns with me merging this from your perspective?

**Comment by hweng** [comment]
@mcritchlow Would you like to squash all those commits, or do you want me to do it?

**Comment by hweng** [comment]
:+1:
